File: id/src/main/java/com/simprints/id/tools/TimeoutBar.java
package com.simprints.id.tools;

import android.content.Context;
import android.graphics.drawable.Drawable;
import android.os.CountDownTimer;
import android.widget.ProgressBar;

import com.simprints.id.R;
import com.simprints.id.domain.Finger;

import androidx.core.content.ContextCompat;

public class TimeoutBar {
    private Context context;
    private CountDownTimer countDownTimer;
    private ProgressBar progressBar;
    private int timeoutMs;

    public TimeoutBar(Context context, ProgressBar progressBar, int timeoutMs) {
        this.context = context;
        this.progressBar = progressBar;
        this.timeoutMs = timeoutMs;
    }

    public void startTimeoutBar() {
        final int[] i = {0, timeoutMs};
        progressBar.setProgress(i[0]);
        countDownTimer = new CountDownTimer(i[1], i[1] / 100) {
            @Override
            public void onTick(long millisUntilFinished) {
                i[0] += 1;
                progressBar.setProgress(i[0]);
            }

            @Override
            public void onFinish() {
                progressBar.setProgress(100);
            }
        };
        countDownTimer.start();
    }

    public void stopTimeoutBar() {
        if (countDownTimer == null)
            return;

        countDownTimer.cancel();
        countDownTimer.onFinish();
    }

    public void cancelTimeoutBar() {
        if (countDownTimer == null)
            return;

        countDownTimer.cancel();
        progressBar.setProgress(0);
    }

    public void setProgressBar(Finger.Status status) {
        progressBar.setProgress(0);
        Drawable drawable;

        switch (status) {
            case NOT_COLLECTED:
                drawable = ContextCompat.getDrawable(context,
                        R.drawable.timer_progress_bar);
                break;

            case GOOD_SCAN:
                drawable = ContextCompat.getDrawable(context,
                        R.drawable.timer_progress_good);
                break;

            case BAD_SCAN:
            case NO_FINGER_DETECTED:
                drawable = ContextCompat.getDrawable(context,
                        R.drawable.timer_progress_bad);
                break;

            default:
                return;

        }

        progressBar.setProgressDrawable(drawable);
    }

}

File: fingerprint/src/main/java/com/simprints/fingerprint/activities/collect/views/TimeoutBar.kt
package com.simprints.fingerprint.activities.collect.views

import android.content.Context
import android.graphics.drawable.Drawable
import android.os.CountDownTimer
import android.widget.ProgressBar

import com.simprints.fingerprint.R
import com.simprints.fingerprint.activities.collect.models.FingerStatus

import androidx.core.content.ContextCompat

class TimeoutBar(private val context: Context, private val progressBar: ProgressBar, private val timeoutMs: Int) {
    private var countDownTimer: CountDownTimer? = null

    fun startTimeoutBar() {
        val i = intArrayOf(0, timeoutMs)
        progressBar.progress = i[0]
        countDownTimer = object : CountDownTimer(i[1].toLong(), (i[1] / 100).toLong()) {
            override fun onTick(millisUntilFinished: Long) {
                i[0] += 1
                progressBar.progress = i[0]
            }

            override fun onFinish() {
                progressBar.progress = 100
            }
        }
        countDownTimer!!.start()
    }

    fun stopTimeoutBar() {
        if (countDownTimer == null)
            return

        countDownTimer!!.cancel()
        countDownTimer!!.onFinish()
    }

    fun cancelTimeoutBar() {
        if (countDownTimer == null)
            return

        countDownTimer!!.cancel()
        progressBar.progress = 0
    }

    fun setProgressBar(status: FingerStatus) {
        progressBar.progress = 0

        val drawable: Drawable? = when (status) {
            FingerStatus.NOT_COLLECTED -> ContextCompat.getDrawable(context, R.drawable.timer_progress_bar)
            FingerStatus.GOOD_SCAN -> ContextCompat.getDrawable(context, R.drawable.timer_progress_good)
            FingerStatus.BAD_SCAN, FingerStatus.NO_FINGER_DETECTED -> ContextCompat.getDrawable(context, R.drawable.timer_progress_bad)
            else -> return
        }

        progressBar.progressDrawable = drawable
    }

}
